modules/site-health/audit-full-page-cache/load.php

```diff
@@ -0,0 +1,198 @@
+			'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+			esc_url( 'https://wordpress.org/support/article/optimization/#Caching' ),
+			esc_html__( 'Learn more about page caching', 'performance-lab' ),
+			/* translators: The accessibility text. */
+			esc_html__( '(opens in a new tab)', 'performance-lab' )
+		),
+	);
+
+	$page_cache_detail = perflab_afpc_get_page_cache_detail();
+
+	if ( is_wp_error( $page_cache_detail ) ) {
+		return perflab_afpc_page_cache_unable_detect_cache_test( $result, $page_cache_detail );
+	}
+
+	return perflab_afpc_page_cache_detected_test( $result, $page_cache_detail );
+}
+
+/**
+ * Response when unable to detect the presence of page caching.
+ *
+ * @param array    $default_response Default perflab_page_cache callback response.
+ * @param WP_Error $page_cache_detail_error get_page_cache_detail() response.
+ *
+ * @return array
+ * @since 1.0.0
+ */
+function perflab_afpc_page_cache_unable_detect_cache_test( $default_response, $page_cache_detail_error ) {
+	$default_response['badge']['color'] = 'red';
+	$default_response['label']          = esc_html__( 'Unable to detect the presence of page caching', 'performance-lab' );
+	$default_response['status']         = 'critical';
+	$error_info                         = sprintf(
+	/* translators: 1 is error message, 2 is error code */
+		esc_html__( 'Unable to detect page caching due to possible loopback request problem. Please verify that the loopback request test is passing. Error: %1$s (Code: %2$s)', 'performance-lab' ),
+		$page_cache_detail_error->get_error_message(),
+		$page_cache_detail_error->get_error_code()
+	);
+	$default_response['description'] = wp_kses_post( "<p>$error_info</p>" ) . $default_response['description'];
+	return $default_response;
+}
+
+/**
+ * Response when page cache detected.
+ *
+ * @param array $default_response Default perflab_page_cache callback response.
+ * @param array $page_cache_detail get_page_cache_detail() response.
+ *
+ * @return array
+ * @since 1.0.0
+ */
+function perflab_afpc_page_cache_detected_test( $default_response, $page_cache_detail ) {
+
+	$default_response['status'] = $page_cache_detail['status'];
+
+	switch ( $page_cache_detail['status'] ) {
+		case 'recommended':
+			$default_response['badge']['color'] = 'orange';
+			$default_response['label']          = esc_html__( 'Page caching is not detected but the server response time is OK', 'performance-lab' );
+			break;
+		case 'good':
+			$default_response['badge']['color'] = 'green';
+			$default_response['label']          = esc_html__( 'Page caching is detected and the server response time is good', 'performance-lab' );
+			break;
+		default:
+			$default_response['badge']['color'] = 'red';
+			$default_response['label']          = esc_html__( 'Page caching is detected but the server response time is still slow', 'performance-lab' );
```

Comment:
Should this line be moved to an `else` statement for the following `if` statement?

Reply:
i don't quite understand what you mean here

Reply:
I mean, instead of:

```php
			$default_response['label']          = __( 'Page caching is detected but the server response time is still slow', 'performance-lab' );
			if ( empty( $page_cache_detail['headers'] ) && ! $page_cache_detail['advanced_cache_present'] ) {
				$default_response['label'] = __( 'Page caching is not detected and the server response time is slow', 'performance-lab' );
			}
```

It can be:

```php
			if ( empty( $page_cache_detail['headers'] ) && ! $page_cache_detail['advanced_cache_present'] ) {
				$default_response['label'] = __( 'Page caching is not detected and the server response time is slow', 'performance-lab' );
			} else {
				$default_response['label'] = __( 'Page caching is detected but the server response time is still slow', 'performance-lab' );
			}
```

Reply:
understood :)